docs/source/format/DissociatedIPC.rst

```diff
@@ -0,0 +1,335 @@
+    form. Based on feedback and usage the protocol definition may change until
+    it is fully standardized.
+
+Rationale
+=========
+
+The :ref:`Arrow IPC format <format-ipc>` describes a protocol for transferring
+Arrow data as a stream of record batches. This protocol expects a continuous
+stream of bytes divided into discrete messages (using a length prefix and
+continuation indicator). Each discrete message consists of two portions:
+
+* A `Flatbuffers`_ header message
+* A series of bytes consisting of the flattened and packed body buffers (some
+  message types, like Schema messages, do not have this section)
+  - This is referred to as the *message body* in the IPC format spec.
+
+For most cases, the existing IPC format as it currently exists is extremely efficient:
+
+* Receiving data in the IPC format allows zero-copy utilization of the body
+  buffer bytes, no deserialization is required to form Arrow Arrays
+* An IPC (Feather) file can be memory-mapped because it is location agnostic
+  and the bytes of the file are exactly what is expected in memory.
+
+However, there are use cases that aren't handled by this:
+
+* Constructing the IPC record batch message requires allocating a contiguous
+  chunk of bytes and copying all of the data buffers into it, packed together
+  back-to-back. It's exceedingly difficult to zero-copy **create** IPC messages.
+* If the Arrow data is located in a shared-memory location, there is no standard
+  way to share the handle to the shared-memory across processes or transports that
+  allow for remote memory accessing.
+* Arrow data located on a non-CPU device (such as a GPU) cannot be sent using
+  Arrow IPC without having to copy the data back to the host device or copying
+  the flatbuffer metadata bytes into device memory.
+  - By the same token, receiving IPC messages into device memory would require
+    performing a copy of the flatbuffer metadata back to the host CPU device. This
+    is due to the fact that the IPC stream interleaves data and metadata across a
+    single stream.
+
+This protocol is intended to attempt to solve these use cases in an efficient manner.
+
+Goals
+-----
+
+* Define a generic protocol for passing Arrow IPC data, not tied to any particular
+  transport, that also allows for utilizing non-CPU device memory, shared memory, and
+  newer "high performance" transports such as `ucx`_ or `libfabric`_.
+* Allow for using :ref:`Flight RPC <flight-rpc>` purely for control flow by separating
+  the stream of IPC metadata from IPC body bytes
+  - This allows for the data in the body to be kept on non-CPU devices (like GPUs)
+    without expensive Device -> Host copies.
+
+Definitions
+-----------
+
+.. glossary::
+
+   IPC Metadata
+       The flatbuffer message bytes that encompass the header of an Arrow IPC message
+  
+   Tag
+       A ``uint64`` value used as an ID for a message. Specific bits can be masked to
```

Comment:
Is it little-endian?
Also, is it really an "id" (which is generally expected to be unique)?

Reply:
it is used for identification and control flow. it's not guaranteed to be unique in the general sense as some protocols use them to identify message *types* rather than identify individual messages. Can you think of a better word to use here instead? since we don't want to give the impression that it is intended to be "unique"

Reply:
Something like "a value that determines how the body of a message is to be interpreted".

Perhaps also add some implementation context, such as:
"""
Some protocols, such as UCX, have built-in support for such tag values and will provide them in CPU memory even though the body of the message may reside on a non-CPU device.
"""